src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantBuildsToBeIncremental.cs

```diff
@@ -32,12 +32,50 @@ public void GenerateBuildRuntimeConfigurationFiles_runs_incrementaly()
             var runtimeConfigDevJsonPath = Path.Combine(outputDirectory, "HelloWorld.runtimeconfig.dev.json");
 
             buildCommand.Execute().Should().Pass();
-            DateTime runtimeConfigDevJsonFirstModifiedTime = new FileInfo(runtimeConfigDevJsonPath).LastWriteTime;
+            DateTime runtimeConfigDevJsonFirstModifiedTime = File.GetLastWriteTimeUtc(runtimeConfigDevJsonPath);
 
             buildCommand.Execute().Should().Pass();
-            DateTime runtimeConfigDevJsonSecondModifiedTime = new FileInfo(runtimeConfigDevJsonPath).LastWriteTime;
+            DateTime runtimeConfigDevJsonSecondModifiedTime = File.GetLastWriteTimeUtc(runtimeConfigDevJsonPath);
 
             runtimeConfigDevJsonSecondModifiedTime.Should().Be(runtimeConfigDevJsonFirstModifiedTime);
         }
+
+        [Fact]
+        public void ResolvePackageAssets_runs_incrementally()
+        { 
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld")
+                .WithSource()
+                .Restore(Log);
+
+            var targetFramework = "netcoreapp1.1";
+            var buildCommand = new BuildCommand(Log, testAsset.TestRoot);
+            var outputDirectory = buildCommand.GetOutputDirectory(targetFramework).FullName;
+            var baseIntermediateOutputDirectory = buildCommand.GetBaseIntermediateDirectory().FullName;
+            var intermediateDirectory = buildCommand.GetIntermediateDirectory(targetFramework).FullName;
+
+            var assetsJsonPath = Path.Combine(baseIntermediateOutputDirectory, "project.assets.json");
+            var assetsCachePath = Path.Combine(intermediateDirectory, "HelloWorld.assets.cache");
+
+            // initial build
+            buildCommand.Execute().Should().Pass();
+            var cacheWriteTime1 = File.GetLastWriteTimeUtc(assetsCachePath);
+
+            // build with no change to project.assets.json
+            buildCommand.Execute().Should().Pass();
+            var cacheWriteTime2 = File.GetLastWriteTimeUtc(assetsCachePath);
+            cacheWriteTime2.Should().Be(cacheWriteTime1);
+
+            // build with modified project
+            File.SetLastWriteTimeUtc(assetsJsonPath, DateTime.UtcNow);
+            buildCommand.Execute().Should().Pass();
+            var cacheWriteTime3 = File.GetLastWriteTimeUtc(assetsCachePath);
+            cacheWriteTime3.Should().NotBe(cacheWriteTime2);
```

Comment:
Is it possible that a build would run fast enough that these times are actually the same?  Should the test do a short sleep if `DateTime.UtcNow` hasn't increased at each stage?

Reply:
Fixed.